workers/scanner.go

```diff
@@ -279,6 +290,27 @@ func (s *Scanner) staticScheduleNextScan() bool {
 func (s *Scanner) managedPinUnderpinnedSkylinks() {
 	s.staticLogger.Trace("Entering managedPinUnderpinnedSkylinks")
 	defer s.staticLogger.Trace("Exiting  managedPinUnderpinnedSkylinks")
+
+	// Clear out the skipped skylinks from the previous run.
+	s.mu.Lock()
+	s.skipSkylinks = []string{}
+	s.mu.Unlock()
+
+	intermediateStatsTicker := time.NewTicker(printPinningStatisticsPeriod)
+	defer intermediateStatsTicker.Stop()
+	countPinned := 0
```

Comment:
nit: might be helpful to also count total or failed so that we know a little more about the performance of pinner and each statement is more informative in isolation.

Reply:
That's hard to pull off right now with the current setup. I can add more fields directly on the scanner struct but that doesn't seem like a good approach. The good news is that all skylinks which fail are added to the skipped array. I'll add the number of skipped skylinks to the output and that will give us more info.